homeassistant/config_entries.py

```diff
@@ -759,13 +771,36 @@ async def async_reload(self, entry_id: str) -> bool:
 
         If an entry was not loaded, will just load.
         """
+        entry = self.async_get_entry(entry_id)
+
+        if entry is None:
+            raise UnknownEntry
+
         unload_result = await self.async_unload(entry_id)
 
-        if not unload_result:
+        if not unload_result or entry.disabled_by:
             return unload_result
 
         return await self.async_setup(entry_id)
 
+    async def async_disable(self, entry_id: str, disabled_by: Optional[str]) -> bool:
```

Comment:
I think that this function name is confusing as it can also be used to enable it again.

Maybe:

```suggestion
    async def async_set_disabled_by(self, entry_id: str, disabled_by: Optional[str]) -> bool:
```